database/sqlite3/sqlite3_test.go

```diff
@@ -181,3 +183,71 @@ func TestMigrateWithDirectoryNameContainsWhitespaces(t *testing.T) {
 	}
 	dt.Test(t, d, []byte("CREATE TABLE t (Qty int, Name string);"))
 }
+
+func TestDbPathOutput(t *testing.T) {
+
+	var pathTests = []struct {
+		name string
+		in   string
+		out  string
+		err  error
+	}{
+		// Test inputs will be passed through `net/url` Parse()
+		// Outputs will be the URLs parsed by dbPathFromURL()
+		//
+		// When designing a test that will fail to be parsed (ie, expect net/url.Parse() to throw an error),
+		// we will match the `out` string as either the prefix or suffix of the error string, otherwise the test will fail.
+
+		// simple path tests - no schema
+		{"simple path (no schema)",
+			"/Path/To/A/DB/file.db", "/Path/To/A/DB/file.db", nil},
+		{"simple path (no schema), with whitespaces",
+			"/Path To/A DB/file name.db", "/Path To/A DB/file name.db", nil},
+
+		// simple path tests - relative
+		{"simple path (relative)",
+			"sqlite3://Path/To/A/DB/file.db", "Path/To/A/DB/file.db", nil},
+		{"simple path (relative), with whitespaces",
+			"sqlite3://Path/To/A DB/file name.db", "Path/To/A DB/file name.db", nil},
+		{"simple path (relative), with invalid host",
+			"sqlite3://Path To/A DB/file name.db", "", nurl.InvalidHostError(" ")},
+
+		// simple path tests - absolute
+		{"simple valid path, no whitespaces",
+			"sqlite3:///Path/To/A/DB/file.db", "/Path/To/A/DB/file.db", nil},
+		{"simple path, with whitespaces",
+			"sqlite3:///Path To/A DB/file name.db", "/Path To/A DB/file name.db", nil},
+
+		// path w/query param tests
+		{"path with whitespaces and query params",
+			"sqlite3:///Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d", "/Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d", nil},
+		{"path with whitespaces and query params that require escaping",
+			"sqlite3:///Path To/A DB/file name.db?aQuery=\"something\"&bQuery=else&c=d", "/Path To/A DB/file name.db?aQuery=%22something%22&bQuery=else&c=d", nil},
+		{"path with whitespaces and query params (including custom query param that should be filtered out)",
+			"sqlite3:///Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d&x-custom-query-param=scrubbed", "/Path To/A DB/file name.db?aQuery=something&bQuery=else&c=d", nil},
+
+		// path with % escaped character tests
+		{"path with % escaped characters",
+			"sqlite3:///Path%20To/A%20DB/file%20name.db", "/Path To/A DB/file name.db", nil},
+		{"path with % escaped characters & escaped query params",
+			"sqlite3:///Path%20To/A%20DB/file%20name.db?aQuery=something%20else&c=d", "/Path To/A DB/file name.db?aQuery=something+else&c=d", nil},
+	}
+
+	for _, tt := range pathTests {
+		t.Run(tt.name, func(t *testing.T) {
+			inputURL, err := nurl.Parse(tt.in)
+			if !errors.Is(err, tt.err) {
+				t.Errorf("`in` string failed to parse into a valid URL with error:  <%v>    expected: <%v>", err, tt.err)
+				return
+			} else if err != nil {
+				// we did encouter an error, but it was expected, silently exit this test case.
```

Comment:
What other errors are we expecting? We should be explicit about the errors we expect in our tests. Unexpected errors should fail the test.